presto-main/src/main/java/com/facebook/presto/execution/buffer/SerializedPage.java

```diff
@@ -49,11 +56,33 @@ public SerializedPage(Slice slice, byte pageCodecMarkers, int positionCount, int
         }
     }
 
+    @ThriftConstructor
+    public SerializedPage(byte[] data, byte pageCodecMarkers, int positionCount, int uncompressedSizeInBytes)
+    {
+        this(Slices.wrappedBuffer(data), pageCodecMarkers, positionCount, uncompressedSizeInBytes);
+    }
+
+    @ThriftField(1)
+    public byte[] getData()
+    {
+        Object base = slice.getBase();
+        checkState(base instanceof byte[], "unexpected serialization type %s", base.getClass());
+        byte[] bytes = (byte[]) base;
+
+        if (slice.isCompact()) {
```

Comment:
Should this above three lines actually happen in this `if` branch? I don't see why we would need to get `base` if the slice is not compact.